Add support for opening cross-organisation PRs

JamesDawson opened the pull request:
Adds a new PowerShell module `Endjin.CodeOps`

JamesDawson committed:
WIP

JamesDawson committed:
WIP

JamesDawson committed:
More WIP

JamesDawson committed:
More WIP - moved functions into a module

MikeEvansLarah committed:
Add install-module step

MikeEvansLarah committed:
Install gh CLI

MikeEvansLarah committed:
Set git config

JamesDawson committed:
Finalised migration script
Added some more tests

JamesDawson committed:
Renamed function to use standard verb, tweaked script parameters

JamesDawson committed:
Merge with origin after rebase

JamesDawson committed:
Re-add inadvertently deleted file

JamesDawson committed:
*-VsProjectPackageReference functions now return a change notification flag

JamesDawson committed:
Update-Repo now expects the 'RepoChanges' scriptblock to return a change notification flag
Added error handling for CLI tool error

JamesDawson committed:
Updated specflow migration script to work with latest Endjin.CodeOps changes
Added our standard root exception handler

JamesDawson committed:
Initial change to pr-autoflow sync script to return a change notification flag to Update-Repo

JamesDawson committed:
Multiple repos can now be specified on a single repo entry and logging tweaks

This will make the config more terse for repos that share the same settings

JamesDawson committed:
Switch gh cli authentication to use environment variable mechansim

JamesDawson committed:
Test whether branch needs to be pushed before opening PR

JamesDawson committed:
Test git credential helper in workflow

JamesDawson committed:
Ensure credential helper bash script is executable

JamesDawson committed:
Merge branch 'feature/syncing-prautoflow' into feature/migrate-specflow-metapackage

JamesDawson committed:
Updated repo config to use terser format option

JamesDawson committed:
Added functions to handle JWT generation and GitHub auth

JamesDawson committed:
Initial testing of GitHub auth code with prautoflow script

JamesDawson committed:
Increase logging

JamesDawson committed:
Add support for creating required labels

JamesDawson committed:
Fixed parameter typo

JamesDawson committed:
Fix label color

JamesDawson committed:
Removed now redundant 'generate_token' step from deploy_pr_autoflow workflow

JamesDawson committed:
Set workflow for specflow migration script

JamesDawson committed:
Removed branch trigger from incomplete 'propogate_templates' workflow

HowardvanRooijen commented on Endjin.CodeOps/Endjin.CodeOps.psd1:
```diff
@@ -0,0 +1,132 @@
+GUID = 'f93ff550-8cac-4de2-9ac1-c500a4bac6fb'
+
+# Author of this module
+Author = 'james'
```
Which one? :D

HowardvanRooijen commented on Endjin.CodeOps/Endjin.CodeOps.psd1:
```diff
@@ -0,0 +1,132 @@
+Author = 'james'
+
+# Company or vendor of this module
+CompanyName = 'Unknown'
```
I'm pretty sure I can guess!

HowardvanRooijen commented on Endjin.CodeOps/README.md:
```diff
@@ -0,0 +1,3 @@
+# Endjin.CodeOps
+
+This PowerShell module contains a variety functions useful for manipulating git repos and .NET solutions.
```
Can we have a few more details!

Is there a license?

JamesDawson replied:
The repo has the standard license - module metadata is now updated and I've added a license badge

JamesDawson committed:
Remove branch trigger from specflow migration workflow

JamesDawson committed:
Improved module metadata and README

JamesDawson committed:
Added license badge

JamesDawson committed:
Fix script name in meta package workflow

JamesDawson committed:
Really remove the 'push' trigger on the metapackage workflow

JamesDawson committed:
Merge remote-tracking branch 'origin' into feature/migrate-specflow-metapackage

JamesDawson committed:
Fixed variable name inconsistency

JamesDawson committed:
Fix the 'running in pester' check
Made module loading more consistent between scripts

JamesDawson committed:
Add failing test for projects with no package dependencies

JamesDawson committed:
Add handling for projects with no package dependencies

Add the reference before the Remove operations to avoid scenario we were remove all package dependencies and leave an empty ItemGroup behind.

JamesDawson committed:
Tweak PR title for specflow migration so it triggers pr-autoflow

JamesDawson committed:
WIP: script to close old dependabot prs

JamesDawson committed:
Refactored template sync script to use Endjin.CodeOps module

JamesDawson committed:
Added remaining orgs and latest settings

SpecFlow and template sync features disabled for the moment

JamesDawson committed:
Refactored closing PRs into a module function

JamesDawson committed:
Refactored pr-autoflow and specflow scripts to include steps to close redundant PRs

JamesDawson committed:
Added vellum to test config

JamesDawson committed:
Merge branch 'feature/workflow-sync-refactor' into feature/migrate-specflow-metapackage

JamesDawson committed:
Enable repo-sync for vellum-dotnet in 'live' repo

JamesDawson committed:
Retire 'close-old-dependabot' script as the functinality now lives in 'sync-pr-autoflow' script

JamesDawson committed:
Improved feedback when closing old PRs

JamesDawson committed:
Merge branch 'master' into feature/migrate-specflow-metapackage

JamesDawson committed:
Add other files required for the build workflow

JamesDawson committed:
Make module loading consistent across scenarios

JamesDawson committed:
Test whether gitversion is working properly

JamesDawson committed:
Refactored 'Get-Repos' to support reading config from a remote repo

JamesDawson committed:
Removed 'close_old_dependabot_prs' workflow

JamesDawson committed:
Attempted fix to recent ‘missing zlib’ error

JamesDawson committed:
Does github let you run sudo?

JamesDawson committed:
Try specific .net core SDK version

JamesDawson committed:
Swap 'mkdir' for 'New-Item'

Resolves x-plat inconsistent behaviour

JamesDawson committed:
Fix issue with test data script being executed on module import

JamesDawson committed:
Fix Pester test failure reporting

JamesDawson committed:
Apply dotnet-setup fix to other workflows

Hopefully this will be temporary.

JamesDawson committed:
Changes to 'Get-RepoConfiguration' to make it easier to use
